src/Magento/FunctionalTestingFramework/Module/MagentoWebDriver.php

```diff
@@ -745,23 +745,24 @@ public function _before(TestInterface $test)
      */
     public function dragAndDrop($source, $target, $xOffset = null, $yOffset = null)
     {
-        if ($xOffset !== null || $yOffset !== null) {
-            $snodes = $this->matchFirstOrFail($this->baseElement, $source);
-            $tnodes = $this->matchFirstOrFail($this->baseElement, $target);
+        $snodes = $this->matchFirstOrFail($this->baseElement, $source);
+        $tnodes = $this->matchFirstOrFail($this->baseElement, $target);
+        $action = new WebDriverActions($this->webDriver);
 
+        if ($xOffset !== null || $yOffset !== null) {
             $targetX = intval($tnodes->getLocation()->getX() + $xOffset);
             $targetY = intval($tnodes->getLocation()->getY() + $yOffset);
 
             $travelX = intval($targetX - $snodes->getLocation()->getX());
             $travelY = intval($targetY - $snodes->getLocation()->getY());
-
-            $action = new WebDriverActions($this->webDriver);
-            $action->moveToElement($snodes)->perform();
-            $action->clickAndHold($snodes)->perform();
-            $action->moveByOffset($travelX, $travelY)->perform();
+            $action->moveToElement($snodes);
+            $action->clickAndHold($snodes);
+            $action->moveByOffset($travelX, $travelY);
             $action->release()->perform();
         } else {
-            parent::dragAndDrop($source, $target);
+            $action->clickAndHold($snodes);
+            $action->moveToElement($tnodes);
+            $action->release($tnodes)->perform();
```

Comment:
Any reason for this change? I remember this issue only applies for dragAndDrop with offset.

Reply:
for chrome 81, separating out dragAndDrop into it's native functions helped resolve couple of issues with drag and drop without offsets too, hence the fix.

Reply:
Interesting to know. Maybe php webdriver need to make the some change?

Reply:
yes, we are on latest version of php webdriver. It's probably page builder specific.